index.js

```diff
@@ -47,29 +45,30 @@ function stripComment(sm) {
   return sm.split(',').pop();
 }
 
-function readFromFileMap(sm, dir) {
-  // NOTE: this will only work on the server since it attempts to read the map file
-
+function readFromFileMap(sm, read) {
   var r = exports.mapFileCommentRegex.exec(sm);
-
   // for some odd reason //# .. captures in 1 and /* .. */ in 2
   var filename = r[1] || r[2];
-  var filepath = path.resolve(dir, filename);
 
   try {
-    return fs.readFileSync(filepath, 'utf8');
+    var sm = read(filename);
+    if (sm != null && typeof sm.catch === 'function') {
+      return sm.catch(throwError);
+    } else {
+      return sm;
+    }
   } catch (e) {
-    throw new Error('An error occurred while trying to read the map file at ' + filepath + '\n' + e);
+    throwError(e);
+  }
+
+  function throwError(e) {
+    throw new Error('An error occurred while trying to read the map file at ' + filename + '\n' + e);
```

Comment:
Have you tried what this prints in case of `Error`?
I tried this in the Node.js repl and it does not include the error stack. Afaik `e.toString()` wouldn't either, but `e.stack` does.

Reply:
To be clear I'm talking about the original Error (`e`) here which contains the most useful info arguably

Reply:
Thanks for pointing that out! I just tested with `e.stack` and it prints the original `throw` site at the top of the stack. I'll switch that out